### ISSUE TITLE
Bad cases detection?

### ISSUE THREAD

**Issue by Kizuna-AII** [SizheAn/PanoHead#21]
Hi, thanks to your sharing of code and pretrained model!
Now I am trying to generate lots of human face pictures with PanoHead, and sometimes I get some bad generation pictures, which you have mentioned in the limitation part in CVPR supplementary material.
I am wondering is it possible to detect and filter bad generation results in an automatic way?

Model checkpoint: easy-khair-180-gpc0.8-trans10-025000.pkl

Thanks!

**Comment by Kizuna-AII** [comment]
Now I simply trained a binary classification network to detect bad cases, and it seems to work well.

**Comment by Zvyozdo4ka** [comment]
@Kizuna-AII HOw did you train your network? Can you share some of your outputs, which your consider as good results?

**Comment by Kizuna-AII** [comment]
> @Kizuna-AII HOw did you train your network? Can you share some of your outputs, which your consider as good results?

Sorry that I am not able to get my previous results at this time. 
I just utilize a pre-trained VGG-19 model and modify the last  classification layer to make it output "good" or "bad". Then I manually labelled about 1000 pictures generated by PanoHead with "good" or "bad" annotation, and fine-tune the modified VGG-19 model with these data.
About the "good results" standard, you can just make a "bad" label if the picture has ANY confusing or bad-quality parts.

**Comment by Zvyozdo4ka** [comment]
@Kizuna-AII 
Thank you for explanation!
I did not that PanoHead generates pictures. Are these pictures of resulting 3d model without  or with texture?

**Comment by Kizuna-AII** [comment]
`gen_samples.py` will generate a picture with a given camera pose, which has texture and background.
